lib/indocker/repositories/clonner.rb

```diff
@@ -9,9 +9,16 @@ def initialize(configuration, logger)
   def clone(session, repository)
     raise ArgumenError.new("only git repositories should be clonned") if !repository.is_git?
 
-    session.exec!("rm -rf #{repository.clone_path} && mkdir -p #{repository.clone_path}")
-    
-    git_command = "git clone -b #{repository.branch} --depth 1 #{repository.remote_url} #{repository.clone_path}"
+    repository_remote_url = session.exec!("mkdir -p #{repository.clone_path} && cd #{repository.clone_path} && git config --get remote.origin.url").stdout_data.chomp
+
+    @logger.debug("server remote_url => #{repository_remote_url.inspect}, indocker remote_url => #{repository.remote_url.inspect}")
+
+    git_command = if repository_remote_url == repository.remote_url
+      "cd #{repository.clone_path} && git add . && git reset HEAD --hard && git checkout #{repository.branch} && git pull --force"
```

Comment:
it would be better to create 2 separate methods, e.g. "pull_existing_repository" and "clone_new_repository", otherwise it's not clear what exactly is going on here.

Reply:
👍